common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -317,22 +353,24 @@ public void applicationRemoved(String appId, boolean cleanupLocalDirs) {
     logger.info("Application {} removed, cleanupLocalDirs = {}", appId, cleanupLocalDirs);
     AppShuffleInfo appShuffleInfo = appsShuffleInfo.remove(appId);
     if (null != appShuffleInfo) {
-      mergedShuffleCleaner.execute(
-        () -> closeAndDeletePartitionFilesIfNeeded(appShuffleInfo, cleanupLocalDirs));
+      submitCleanupTask(
+        () -> closeAndDeletePartitions(appShuffleInfo, cleanupLocalDirs, true));
     }
+    removeAppAttemptPathInfoFromDB(
+        new AppAttemptId(appShuffleInfo.appId, appShuffleInfo.attemptId));
```

Comment:
We are relying on DB being consistent with `appsShuffleInfo` map.
Given this, we should move the removal of the DB entry  to be atomic w.r.t `appsShuffleInfo.remove`.

Something like:
```
AtomicReference<AppShuffleInfo> ref = new AtomicReference<>(null);
appsShuffleInfo.compute( appId, (id, info) -> {
    if (null != info) {
      removeAppAttemptPathInfoFromDB(new AppAttemptId(info.appId, info.attemptId));
      ref.set(info);
    }
    // always remove
    return null;
  } );
AppShuffleInfo appShuffleInfo = ref.get
```

Reply:
Updated as suggested